test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts

```diff
@@ -125,14 +125,20 @@ describe('CSOT spec prose tests', function () {
       let childProcess: ChildProcess;
 
       beforeEach(async function () {
-        childProcess = spawn('mongocryptd', ['--port', mongocryptdTestPort, '--ipv6'], {
-          stdio: 'ignore',
-          detached: true
-        });
+        childProcess = spawn(
+          'mongocryptd',
+          ['--port', mongocryptdTestPort, '--ipv6', '--pidfilepath', new ObjectId().toHexString()],
```

Comment:
Adding a random `pidfilepath` ensure it doesn't use the default `mongod` pid path which can cause it to randomly fail